src/d/actor/d_a_npc_roten.cpp

```diff
@@ -5,322 +5,2883 @@
+                offset.z = (*calc_mtx)[2][3];
+                pItem->setOffsetPos(offset);
+                pItem->show();
+                field_0x714 = offset;
+            }
+            else if(field_0x9C0 == 8) {
+                field_0x708 = field_0x714;
+                field_0x99C = 1;
+                field_0x9C4 = 1;
+                field_0x9C6 = 0;
+                field_0x9C7 = 1;
+            }
+        }
+    }
 }
 
 /* 000025FC-00002600       .text eventInit__12daNpcRoten_cFv */
 void daNpcRoten_c::eventInit() {
-    /* Nonmatching */
+    return;
 }
 
 /* 00002600-0000280C       .text eventMesSetInit__12daNpcRoten_cFi */
-void daNpcRoten_c::eventMesSetInit(int) {
-    /* Nonmatching */
+void daNpcRoten_c::eventMesSetInit(int staffIdx) {
+    u32* pData = dComIfGp_evmng_getMyIntegerP(staffIdx, "MsgNo");
+    if(pData) {
+        u32 msgNo = *pData;
+        switch(msgNo) {
+            case 0x0:
+                field_0x98C = l_msg_xy_koukan_item[mNpcNo][field_0x9BE];
+                setMessage(field_0x98C[0]);
+                dComIfGs_onEventBit(l_save_dat[mNpcNo].field_0x06);
+                break;
+            case 0x1:
+                field_0x98C = l_msg_xy_koukan_item2[mNpcNo][field_0x9BE];
+                setMessage(field_0x98C[0]);
+                break;
+            case 0xA:
+                field_0x98C = l_msg_xy_koukan_item3[mNpcNo][field_0x9BE];
+                setMessage(field_0x98C[0]);
+                break;
+            case 0x14:
+                field_0x98C = l_msg_xy_after_get_demo[mNpcNo];
+                setMessage(field_0x98C[0]);
+                break;
+            case 0x15:
+                field_0x98C = l_msg_xy_after_get_demo2[mNpcNo];
+                setMessage(field_0x98C[0]);
+                break;
+            case 0x63:
+                setMessage(getMsg());
+                break;
+            default:
+                setMessage(msgNo);
+                break;
+        }
+    }
+    else {
+        field_0x98C = &field_0x98C[1];
+        setMessage(field_0x98C[0]);
+    }
 }
 
 /* 0000280C-00002840       .text eventMesSet__12daNpcRoten_cFv */
-void daNpcRoten_c::eventMesSet() {
-    /* Nonmatching */
+bool daNpcRoten_c::eventMesSet() {
+    return talk(0) == fopMsgStts_BOX_CLOSED_e;
 }
 
 /* 00002840-000028C4       .text eventSetItemInit__12daNpcRoten_cFv */
 void daNpcRoten_c::eventSetItemInit() {
-    /* Nonmatching */
+    u8 itemIdx = l_item_dat[mNpcNo][field_0x9BE];
+    cXyz pos(0.0f, 0.0f, 0.0f);
+    field_0x6F8 = fopAcM_createItemForPresentDemo(&pos, (u8)(itemIdx + FLOWER_1), 9, -1, fopAcM_GetRoomNo(this)); // fakematch with the cast?
 }
 
 /* 000028C4-0000290C       .text eventSetItem__12daNpcRoten_cFv */
-void daNpcRoten_c::eventSetItem() {
-    /* Nonmatching */
+bool daNpcRoten_c::eventSetItem() {
+    fopAc_ac_c* pActor;
+    if(fopAcM_SearchByID(field_0x6F8, &pActor)) {
+        return pActor; // != NULL gets rid of the clrlwi
```

Comment:
this might work
```suggestion
    if (pActor != NULL) {
        return true;
    } else {
        return false;
    }
```
if/else return true/false produces different codegen than just returning the condition itself, and *also* different from ternary `return (condition) ? true : false;`